db/database.go

```diff
@@ -100,3 +109,65 @@ func (db *Database) ClearAllOffers() error {
 
 	return nil
 }
+
+// PutSwap puts the given swap in the database.
+func (db *Database) PutSwap(s *swap.Info) error {
+	val, err := json.Marshal(s)
+	if err != nil {
+		return err
+	}
+
+	key := s.ID
+	return db.swapTable.Put(key[:], val)
+}
+
+// HasSwap returns whether the db contains a swap with the given ID.
+func (db *Database) HasSwap(id types.Hash) (bool, error) {
+	return db.swapTable.Has(id[:])
+}
+
+// GetSwap returns a swap with the given ID, if it exists.
+func (db *Database) GetSwap(id types.Hash) (*swap.Info, error) {
+	value, err := db.swapTable.Get(id[:])
+	if err != nil {
+		return nil, err
+	}
+
+	var s swap.Info
+	err = json.Unmarshal(value, &s)
+	if err != nil {
+		return nil, err
+	}
+
+	return &s, nil
+}
+
+// GetAllSwaps returns all swaps in the database.
+func (db *Database) GetAllSwaps() ([]*swap.Info, error) {
+	iter := db.swapTable.NewIterator()
+	defer iter.Release()
+
+	swaps := []*swap.Info{}
```

Comment:
The linter in my IDE complains when we do declarations like this.  It recommends:
```
var offers []*types.Offer
```
`nil` plays nice with `append` and wouldn't break anything in this function (or most good go code), but if you are worried that someone calling this function might not be treating an empty slice and `nil` the same way, the linter also likes this:
```
offers := make([]*types.Offer, 0)
```